mne/viz/topomap.py

```diff
@@ -2382,3 +2382,86 @@ def _plot_corrmap(data, subjs, indices, ch_type, ica, label, show, outlines,
     fig.canvas.draw()
     plt_show(show)
     return fig
+
+
+def _trigradient(x, y, z):
+    """Take gradients of z on a mesh."""
+    from matplotlib.tri import CubicTriInterpolator, Triangulation
+    tri = Triangulation(x, y)
+    tci = CubicTriInterpolator(tri, z)
+    dx, dy = tci.gradient(tri.x, tri.y)
```

Comment:
I thought it was better before with the `with warnings.catch_warnings()` here instead of in the tests, that way users will not get `DeprecationWarnings` during use that are due to upstream lib bugs

Reply:
fixed